src/segmantic/seg/monai_unet.py

```diff
@@ -482,26 +472,26 @@ def train(
             else:
                 plt.show()
 
-            if output_dir:
+            if saver:
                 pred_labels = val_outputs.argmax(dim=1, keepdim=True)
                 saver.save_batch(pred_labels, val_data["image_meta_dict"])
 
 
 def predict(
-    model_file: str,
-    test_images: List[str],
-    test_labels: List[str] = None,
+    model_file: Path,
+    output_dir: Path,
+    test_images: List[Path],
+    test_labels: List[Path] = None,
     tissue_dict: Dict[str, int] = None,
-    output_dir: str = None,
     save_nifti: bool = False,
     gpu_ids: list = [],
 ):
     # load trained model
-    with open(model_file.rsplit(".", 1)[0] + ".json") as json_file:
+    with open(os.path.splitext(model_file)[0] + ".json") as json_file:
```

Comment:
it is worth having a look to the [``pathlib.Path``](https://docs.python.org/3/library/pathlib.html) which is a very readable and convenient way to deal with paths. For instance, to change the extension of a path it would be as

```python
from pathlib import Path

model_file: Path

with model_file.with_suffix(".json").open() as json_file:
    ...

```